src/tracking/envelope.cpp

```diff
@@ -147,14 +148,27 @@ namespace impactx
                                        << " slice_step=" << slice_step << "\n";
                     }
 
+                    std::visit([&ref, &cm](auto&& element)
```

Comment:
Is the reference particle pushed twice now? o.0

Happens again in line 176 below

Reply:
I think this is responsible for the failing tests